cfdm/read_write/netcdf/netcdfwrite.py

```diff
@@ -5289,3 +5310,83 @@ def _customise_write_vars(self):
 
         """
         pass
+
+    def _chunking_parameters(self, data, ncdimensions):
+        """Set chunking parameters for `netCDF4.createVariable`.
+
+        .. versionadded:: (cfdm) NEXTVERSION
+
+        :Parameters:
+
+            data: `Data` or `None`
+                The data being written.
+
+            ncdimensions: `tuple`
+                The data netCDF dimensions.
+
+        :Returns:
+
+            2-tuple
+                The *contiguous* and *chunksizes* parameters for
+                `netCDF4.createVariable`.
+
+        """
+        if data is None:
+            return False, None
+
+        g = self.write_vars
+
+        # ------------------------------------------------------------
+        # HDF5 chunk strategy: Either use that provided on the data,
+        # or else work it out.
+        # ------------------------------------------------------------
+        # Get the chunking strategy defined by the data itself
+        chunksizes = self.implementation.nc_get_hdf5_chunksizes(data)
+        if chunksizes == "contiguous":
+            # Contiguous as defined by 'data'
+            return True, None
+
+        # Still here?
+        hdf5_chunks = g["hdf5_chunks"]
+        if isinstance(chunksizes, int):
+            # Reset hdf_chunks to the integer given by 'data'
+            hdf5_chunks = chunksizes
+        elif chunksizes is not None:
+            # Chunked as defined by the tuple of int given by 'data'
+            return False, chunksizes
+
+        # Still here? Then work out the chunking strategy from the
+        # hdf5_chunnks
+        if hdf5_chunks == "contiguous":
+            # Contiguous as defined by 'hdf_chunks'
+            return True, None
+
+        # Still here? Then work out the chunks from both the
+        # size-in-bytes given by hdf5_chunks (e.g. 1024, or '1 KiB'),
+        # and the data shape (e.g. (12, 73, 96)).
+        compressed = bool(
+            set(ncdimensions).intersection(g["sample_ncdim"].values())
+        )
```

Comment:
Seems like a slightly contrived way to determine if the data is compressed (if I understand the aim of the variable correctly), can/should we instead use a method designed to test this sort of thing e.g:

```suggestion
        compressed = data.get_compression_type() != ""
```
?

Reply:
I think something like that would be much nicer! ... However, data this is actually compressed might be being written out uncompressed, and that's why we test for compression on the actual ncdimensions in the output ffile, rather than the `Data` object itself.

Moving that nasty line to a method is a good idea. How about: c16d212bffb243d5c5ab9de843fee28c80b4a289

Reply:
All of these horrors will magically disappear when we refactor `netcdfwrite.py` :)

Reply:
Ah, I see! In which case, all good, though good idea with moving it out to a method to clarify the nature and in case we want to use that elsewhere.

Reply:
The new method looks good but there's a few typos (see my pair of new suggestions), once those are fixed I think we are ready to merge if you agree :)